grammar.js

```diff
@@ -998,6 +1024,148 @@ module.exports = grammar({
       'end',
     ),
 
+    lib_def: $ => seq(
+      'lib',
+      field('name', $.constant, $.generic_type),
+      optional($._lib_statements),
+      'end'
+    ),
+
+    top_level_fun_def: $ => {
+      const params = seq(
+        '(', field('params', $.fun_param_list), ')',
```

Comment:
Params are optional within the parentheses. Also, `fun_param_list` doesn't need to be a distinct node type in the final syntax tree. We can alias `fun_param_list` -> `param_list`, `fun_param` -> `param`, etc.
```suggestion
        '(', optional(field('params', alias($.fun_param_list, $.param_list))), ')',
```